homeassistant/components/ozw/light.py

```diff
@@ -140,29 +140,18 @@ def color_temp(self):
     def async_set_duration(self, **kwargs):
         """Set the transition time for the brightness value.
 
-        Zwave Dimming Duration values:
-        0       = instant
-        0-127   = 1 second to 127 seconds
-        128-254 = 1 minute to 127 minutes
-        255     = factory default
+        Zwave Dimming Duration values now use seconds as an
+        integer (max: 7620 seconds or 127 mins)
         """
         if self.values.dimming_duration is None:
             return
 
         if ATTR_TRANSITION not in kwargs:
             # no transition specified by user, use defaults
-            new_value = 255
+            new_value = 7621  # anything over 7620 uses the factory default
```

Comment:
We need to add a compatibility layer that checks the version of OpenZWave and applies the correct logic so we're backwards compatible.

We can use this topic:
`OpenZWave/<instance>/status`
```
{
    "OpenZWave_Version": "1.6.1198",
    "OZWDaemon_Version": "0.1.0",
    "QTOpenZWave_Version": "1.2.0",
    "QT_Version": "5.12.5",
    "Status": "driverAllNodesQueried",
    "TimeStamp": 1595274687,
    "ManufacturerSpecificDBReady": true,
    "homeID": 3615043505,
    "getControllerNodeId": 1,
    "getSUCNodeId": 0,
    "isPrimaryController": false,
    "isBridgeController": false,
    "hasExtendedTXStatistics": false,
    "getControllerLibraryVersion": "Z-Wave 4.05",
    "getControllerLibraryType": "Static Controller",
    "getControllerPath": "/dev/serial/by-id/usb-Silicon_Labs_HubZ_Smart_Home_Controller_C0F009CE-if00-port0"
}
```

Reply:
How can I get to that info? `self.value.primary.node.values().instance.status`?

Reply:
We can get the status instance like this:
https://github.com/home-assistant/core/blob/36cb818cd0020b0bc060f4500cb36f3c8d412b22/homeassistant/components/ozw/entity.py#L218

Then use this property:
https://github.com/cgarwood/python-openzwave-mqtt/blob/ede4f28e739045d6cf25d7268a9b356a3cabacbd/openzwavemqtt/models/instance_status.py#L31-L34

Reply:
Guess I was kinda close, thank you, I'll get it worked out.

Reply:
I think the version you're looking for is `1.6.1205`. This is the commit that changed the duration from z-wave encoding to integer seconds.
```
$ git log -n1
commit f81bc04b672d52b7bd5f8e8b604058c170c9911b (HEAD)
Author: Justin Hammond <Justin@dynam.ac>
Date:   Fri Jul 10 00:40:02 2020 +0800

    Duration Encoding Fixes and Factor in Default Duration Setting

$ git describe --long --tags --dirty
v1.6-1205-gf81bc04b
```

Reply:
Going with build number higher than what came with build-150.

Reply:
@MartinHjelmare version checks in the latest commit

Reply:
The addon v0.5.1 uses v1.6-1198-g6c2ca613.

Reply:
Addon v0.5.0 uses v1.6-1201-g5d3978d5. :)

Reply:
Those all are higher than 1185 and had the new dimmer code in them which was way the 0.5.1 was rolled back, along with a few other nasty reasons.

Reply:
> Those all are higher than 1185

Versions < 1205 use the old dim duration encoding. So the question is what are you trying to be backwards compatible with?

0.5.2 is v1.6-1210-g6cf37295, has new dim duration
0.5.1 is v1.6-1198-g6c2ca613, has old dim duration. Reverted back to QT 5.12 and pre-dimmer refresh changes.
0.5.0 is v1.6-1201-g5d3978d5, has old dim duration. Updated to QT 5.14 (which had problems). Contains some broken dimmer refresh code, but not the duration type change. This is a botched release.
0.4.4 is v1.6-1149-ga35732f1, has old dim duration, has QT 5.12

docker build-160 is v1.6-1210-g6cf37295, has new dim duration. Builds 160+ of course have the new dim duration.
docker build-152 is v1.6-1197-g2f2342a7, has dim old duration
docker build-151 is v1.6-1185-g0dba812c, has dim old duration
docker build-150 is v1.6-1185-g0dba812c, has dim old duration

1185 will work for 0.4.4 and 0.5.2 addon users, but not 0.5.1 (nor 0.5.0). It will also work for docker users with builds 151 and earlier, and build 160 and up. Most docker users I'd guess are just blindly pulling latest.

Reply:
Ok I'll up the build, easy fix :)